server/sonar-server/src/main/java/org/sonar/server/db/migrations/AddColumnsBuilder.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.db.migrations;
+
+import org.sonar.core.persistence.dialect.Dialect;
+import org.sonar.core.persistence.dialect.MsSql;
+import org.sonar.core.persistence.dialect.Oracle;
+import org.sonar.core.persistence.dialect.PostgreSql;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+public class AddColumnsBuilder {
+
+  private final Dialect dialect;
+  private final String tableName;
+  private List<ColumnDef> columnDefs = newArrayList();
+
+  public AddColumnsBuilder(Dialect dialect, String tableName) {
+    this.tableName = tableName;
+    this.dialect = dialect;
+  }
+
+  public AddColumnsBuilder addColumn(ColumnDef columnDef) {
+    columnDefs.add(columnDef);
+    return this;
+  }
+
+  public String build() {
+    StringBuilder sql = new StringBuilder().append("ALTER TABLE ").append(tableName).append(" ");
+    addSql(sql);
+    return sql.toString();
+  }
+
+  private void addSql(StringBuilder sql){
+    switch (dialect.getId()) {
+      case PostgreSql.ID:
+        addColumns(sql, "ADD COLUMN ");
+        break;
+      case MsSql.ID:
+        sql.append("ADD ");
+        addColumns(sql, null);
+        break;
+      default:
+        sql.append("ADD (");
+        addColumns(sql, null);
+        sql.append(")");
+    }
+  }
+
+  private void addColumns(StringBuilder sql, @Nullable String columnPrefix) {
```

Comment:
Nullable parameters should be avoided as much as possible. I would replace null by empty "". It also removes the condition if (columnPrefix != null)